summer-of-code/week-02/wk2-homework-submissions/soc_wk2_cert_cristina_tarantino.test.py

```diff
@@ -0,0 +1,542 @@
+
+    def test_char_frequency_list_from_alphabet_file(self):
+        filename = "../alice_in_wonderland.txt"
+        # source http://www.gutenberg.org/files/11/11-0.txt
+
+        # open the file in read only mode
+        # source https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
+        with open(filename) as f:
+            read_data = f.read()
+        f.closed
+
+        expected_result = [['a', 9804], ['b', 1746], ['c', 3003], ['d', 5469], ['e', 15396], ['f', 2383], ['g', 2944],
+                           ['h', 7890], ['i', 8634], ['j', 235], ['k', 1290], ['l', 5211], ['m', 2466], ['n', 8053],
+                           ['o', 9480], ['p', 1968], ['q', 220], ['r', 6612], ['s', 7269], ['t', 12204], ['u', 3979],
+                           ['v', 963], ['w', 2952], ['x', 176], ['y', 2584], ['z', 80]]
+
+        self.assertEqual(homework.char_frequency_list_from_alphabet(read_data), expected_result)
+
+
+class TestCharFrequencyListOnlyPresentAlphas(unittest.TestCase):
+    def test_char_frequency_dict_only_present_alphas_string(self):
+        test_text = "Ciao io sono Cristina"
+        expected_result = [['a', 2], ['c', 2], ['i', 4], ['n', 2], ['o', 4], ['r', 1], ['s', 2], ['t', 1]]
+
+        self.assertEqual(homework.char_frequency_list_only_present_alphas(test_text), expected_result)
+
+    def test_char_frequency_dict_only_present_alphas_file(self):
+        filename = "../alice_in_wonderland.txt"
+        # source http://www.gutenberg.org/files/11/11-0.txt
+
+        # open the file in read only mode
+        # source https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
+        with open(filename) as f:
+            read_data = f.read()
+        f.closed
+
+        expected_result = [['a', 9804], ['b', 1746], ['c', 3003], ['d', 5469], ['e', 15396], ['f', 2383], ['g', 2944],
+                           ['h', 7890], ['i', 8634], ['j', 235], ['k', 1290], ['l', 5211], ['m', 2466], ['n', 8053],
+                           ['o', 9480], ['p', 1968], ['q', 220], ['r', 6612], ['s', 7269], ['t', 12204], ['u', 3979],
+                           ['v', 963], ['w', 2952], ['x', 176], ['y', 2584], ['z', 80]]
+
+        self.assertEqual(homework.char_frequency_list_only_present_alphas(read_data), expected_result)
+
+
+# https://learnbyexample.gitbooks.io/python-basics/content/Testing.html#using-unittest.mock-to-test-user-input-and-program-output
+# TODO research decorator and sys.stdout
+class TestNumbersToLetters(unittest.TestCase):
+    @mock.patch('sys.stdout', new_callable=StringIO)
+    def test_numbers_to_letters(self, mock_stdout):
+        expected_result = 'A\nB\nC\nD\nE\nF\nG\nH\nI\nJ\nK\nL\nM\nN\nO\nP\nQ\nR\nS\nT\nU\nV\nW\nX\nY\nZ\na\nb\nc\nd\ne\nf\ng\nh\ni\nj\nk\nl\nm\nn\no\np\nq\nr\ns\nt\nu\nv\nw\nx\ny\nz\n'
+        homework.numbers_to_letters()
+        self.assertEqual(mock_stdout.getvalue(), expected_result)
+
+
```

Comment:
Its best to have comments before test cases describing what the test case actually does or trying to find out.